lib/fs/basicfs_case.go

```diff
@@ -0,0 +1,409 @@
+		return err
+	}
+	if err := f.BasicFilesystem.Rename(oldpath, newpath); err != nil {
+		return err
+	}
+	f.cleanCase()
+	return nil
+}
+
+func (f *caseBasicFilesystem) Stat(name string) (FileInfo, error) {
+	var err error
+	if name, err = Canonicalize(name); err != nil {
+		return nil, err
+	}
+	stat, err := f.BasicFilesystem.Stat(name)
+	if err != nil {
+		l.Infoln("herarerewrfs", name)
+		return nil, err
+	}
+	realName, err := f.realCase(name)
+	if err != nil {
+		return nil, err
+	}
+	if realName != name {
+		return nil, &ErrCase{name, realName}
+	}
+	return stat, nil
+}
+
+func (f *caseBasicFilesystem) DirNames(name string) ([]string, error) {
+	if err := f.checkCase(name); err != nil {
+		return nil, err
+	}
+	return f.BasicFilesystem.DirNames(name)
+}
+
+func (f *caseBasicFilesystem) Open(name string) (File, error) {
+	if err := f.checkCase(name); err != nil {
+		return nil, err
+	}
+	return f.BasicFilesystem.Open(name)
+}
+
+func (f *caseBasicFilesystem) OpenFile(name string, flags int, mode FileMode) (File, error) {
+	if err := f.checkCase(name); err != nil {
+		return nil, err
+	}
+	file, err := f.BasicFilesystem.OpenFile(name, flags, mode)
+	if err != nil {
+		return nil, err
+	}
+	f.cleanCase()
+	return file, nil
+}
+
+func (f *caseBasicFilesystem) ReadSymlink(name string) (string, error) {
+	if err := f.checkCase(name); err != nil {
+		return "", err
+	}
+	return f.BasicFilesystem.ReadSymlink(name)
+}
+
+func (f *caseBasicFilesystem) Create(name string) (File, error) {
+	if err := f.checkCase(name); err != nil {
+		return nil, err
+	}
+	file, err := f.BasicFilesystem.Create(name)
+	if err != nil {
+		return nil, err
+	}
+	f.cleanCase()
+	return file, nil
+}
+
+func (f *caseBasicFilesystem) CreateSymlink(target, name string) error {
+	if err := f.checkCase(name); err != nil {
+		return err
+	}
+	if err := f.BasicFilesystem.CreateSymlink(target, name); err != nil {
+		return err
+	}
+	f.cleanCase()
+	return nil
+}
+
+func (f *caseBasicFilesystem) Walk(root string, walkFn WalkFunc) error {
+	if err := f.checkCase(root); err != nil {
+		return err
+	}
+	return f.BasicFilesystem.Walk(root, walkFn)
+}
+
+func (f *caseBasicFilesystem) Watch(path string, ignore Matcher, ctx context.Context, ignorePerms bool) (<-chan Event, <-chan error, error) {
+	if err := f.checkCase(path); err != nil {
+		return nil, nil, err
+	}
+	return f.BasicFilesystem.Watch(path, ignore, ctx, ignorePerms)
+}
+
+func (f *caseBasicFilesystem) Hide(name string) error {
+	if err := f.checkCase(name); err != nil {
+		return err
+	}
+	return f.BasicFilesystem.Hide(name)
+}
+
+func (f *caseBasicFilesystem) Unhide(name string) error {
+	if err := f.checkCase(name); err != nil {
+		return err
+	}
+	return f.BasicFilesystem.Unhide(name)
+}
+
+type caseBasicFile struct {
+	basicFile
+	caseFS *caseBasicFilesystem
+}
+
+func (f caseBasicFile) Stat() (FileInfo, error) {
+	if err := f.caseFS.checkCase(f.name); err != nil {
+		return nil, err
+	}
+	return f.basicFile.Stat()
+}
```

Comment:
Not used. I guess it's needed if the file case changes after opening, yet I wonder if stat on a normal filesystem fails if the file is renamed, or if it's done on the file descriptor.